### PR TITLE
Implement getter for child BCs correctly

### LINKED ISSUE #20890: GetChildBrowsingContextId returns BCIDs in reverse tree-order
Given the following HTML:
```
<!doctype html>
<title>cross-document channel</title>
<iframe src=004-1.html></iframe>
<iframe src=004-2.html></iframe>
```

Running `window[0]` would return the iframe with the src of `004-2.html`, and `window[1]` returns the one with `004-1.html`.

The relevant code snippet in the constellation that processes this IPC message is here:
https://github.com/servo/servo/blob/022daccbe102aa67c6a4442f8d9cbccc3a7e221b/components/constellation/constellation.rs#L1166-L1174

... which causes me to think that perhaps the `AllBrowsingContextsIterator` defined in `constellation/browsingcontext.rs` is traversing the HTML tree in reversed tree order... but I think that isn't really the case either, so perhaps the HTML parser is inserting them in reversed order?

### PR DESCRIPTION
**Pull request by KiChjang**
Fixes #20890.

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/20891)

### CODE REVIEW

**Comment by KiChjang** [comment]
I've [talked with gsnedders on #whatwg](https://freenode.logbot.info/whatwg/20180531#c1568098), and he confirms that the window indexed getter only goes 1 level deep (i.e. the grandchildren BCs will not be included), so this should produce the correct order.

@bors-servo try

File: components/constellation/constellation.rs [hunk 1]
```diff
@@ -1165,7 +1165,7 @@ impl<Message, LTF, STF> Constellation<Message, LTF, STF>
             }
             FromScriptMsg::GetChildBrowsingContextId(browsing_context_id, index, sender) => {
                 // We increment here because the 0th element is the parent browsing context itself
-                let result = self.all_descendant_browsing_contexts_iter(browsing_context_id)
+                let result = self.fully_active_descendant_browsing_contexts_iter(browsing_context_id)
                     .nth(index + 1)
```
Comment (asajeffrey) [inline components/constellation/constellation.rs]: This seems odd, if nothing else our naming is out of sync. Is this intended to return the i'th child browsing context or the i'th descendant?

Comment (KiChjang) [reply components/constellation/constellation.rs]: The ith child, because grandchildren and their offspring does not need to be included.

Comment (KiChjang) [reply components/constellation/constellation.rs]: Oh, I see what's wrong with the naming. I'll double check all the use cases and see if they're indeed all simply looking at the children only and not their descendants as well.

**Comment by KiChjang** [comment]
@asajeffrey That was a good catch! It looks like the child BC getter implementation did not even require the usage of the BC iterators, and instead I can just look at the children field of the corresponding pipeline and get its child BCs from there.

@bors-servo try

**Comment by KiChjang** [comment]
@bors-servo retry
* Stack overflow?!

**Comment by jdm** [comment]
@emilio The stack overflow continues to plague us :((

**Pull request merged**